is.ts

```diff
@@ -17,1344 +23,2625 @@ import type {
-      assertType(is.integer(value), AssertionTypeDescription.integer, value),
-    safeInteger: (value: unknown): asserts value is number =>
-      assertType(
-        is.safeInteger(value),
-        AssertionTypeDescription.safeInteger,
-        value,
-      ),
-    evenInteger: (value: number): asserts value is number =>
-      assertType(
-        is.evenInteger(value),
-        AssertionTypeDescription.evenInteger,
-        value,
-      ),
-    oddInteger: (value: number): asserts value is number =>
-      assertType(
-        is.oddInteger(value),
-        AssertionTypeDescription.oddInteger,
-        value,
-      ),
-    infinite: (value: unknown): asserts value is number =>
-      assertType(is.infinite(value), AssertionTypeDescription.infinite, value),
-    numericString: (value: unknown): asserts value is string =>
-      assertType(
-        is.numericString(value),
-        AssertionTypeDescription.numericString,
-        value,
-      ),
-    inRange: (
-      value: number,
-      range: number | number[],
-    ): asserts value is number =>
-      assertType(
-        is.inRange(value, range),
-        AssertionTypeDescription.inRange,
-        value,
-      ),
+    if (isDeprecated(target, p) || is.symbol(p)) {
+      return { ...desc, enumerable: false };
+    }
+    return desc;
+  },
+}) as is;
 
-    emptyArray: (value: unknown): asserts value is never[] =>
-      assertType(
-        is.emptyArray(value),
-        AssertionTypeDescription.emptyArray,
-        value,
-      ),
-    emptySet: (value: unknown): asserts value is Set<never> =>
-      assertType(is.emptySet(value), AssertionTypeDescription.emptySet, value),
-    emptyMap: (value: unknown): asserts value is Map<never, never> =>
-      assertType(is.emptyMap(value), AssertionTypeDescription.emptyMap, value),
-    emptyObject: <Key extends keyof any = string>(
-      value: unknown,
-    ): asserts value is Record<Key, never> =>
-      assertType(
-        is.emptyObject(value),
-        AssertionTypeDescription.emptyObject,
-        value,
-      ),
-    emptyString: (value: unknown): asserts value is "" =>
-      assertType(
-        is.emptyString(value),
-        AssertionTypeDescription.emptyString,
-        value,
-      ),
-    whitespace: (value: unknown): asserts value is string =>
-      assertType(
-        is.whitespace(value),
-        AssertionTypeDescription.whitespace,
-        value,
-      ),
-    emptyStringOrWhitespace: (value: unknown): asserts value is string =>
-      assertType(
-        is.emptyStringOrWhitespace(value),
-        AssertionTypeDescription.emptyStringOrWhitespace,
-        value,
-      ),
+/**
+ * Determine the {@linkcode TypeName} of an arbitrary value of unknown type.
+ *
+ * @example
+ * ```ts
+ * import { is } from "https://deno.land/x/dis/mod.ts";
+ *
+ * getTypeName("🦕") // => "string"
+ * getTypeName(100n) // => "bigint"
+ * getTypeName({ foo: "bar" }) // => "Object"
```

Comment:
Moved the "function" check out of this block to allow for objects of type "AsyncFunction" and "Class" to be identified first.